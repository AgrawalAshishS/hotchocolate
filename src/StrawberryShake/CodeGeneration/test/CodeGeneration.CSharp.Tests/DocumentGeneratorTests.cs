using System.Text;
using System.Threading.Tasks;
using Snapshooter.Xunit;
using Xunit;

namespace StrawberryShake.CodeGeneration.CSharp
{
    public class DocumentGeneratorTests
    {
        [Fact]
        public async Task GenerateModel()
        {
            // arrange
            var sb = new StringBuilder();
            var writer = new CodeWriter(sb);

            var generator = new DocumentGenerator();

            var descriptor = new DocumentDescriptor(
                "Queries",
                "Demo",
                new byte[] { 1,2 ,3 },
                new byte[] { 4,5 ,6 },
                new byte[] { 7,8 ,9 },
                @"type Query {
                    s: String
                }");

            // act
            await generator.WriteAsync(writer, descriptor);

            // assert
            sb.ToString().MatchSnapshot();
        }

        [Fact]
        public void CanHandle()
        {
            // arrange
            var generator = new DocumentGenerator();

            var descriptor = new DocumentDescriptor(
                "Queries",
                "Demo",
                new byte[] { 1,2 ,3 },
                new byte[] { 4,5 ,6 },
                new byte[] { 7,8 ,9 },
                @"type Query {
                    s: String
                }");

            // act
            var canHandle = generator.CanHandle(descriptor);

            // assert
            Assert.True(canHandle);
        }
    }
}
