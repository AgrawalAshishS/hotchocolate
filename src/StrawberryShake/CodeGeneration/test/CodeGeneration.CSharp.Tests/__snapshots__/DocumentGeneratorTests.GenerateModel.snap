[global::System.CodeDom.Compiler.GeneratedCode("StrawberryShake", "11.0.0")]
public partial Queries
    : global::StrawberryShake.IDocument
{
    private readonly byte[] _hashName = new byte[]
{
    1,
    2,
    3
};
    private readonly byte[] _hash = new byte[]
{
    4,
    5,
    6
};
    private readonly byte[] _content = new byte[]
{
    7,
    8,
    9
};

    public global::System.ReadOnlySpan<byte> HashName
    {
        get => _hashName
    }

    public global::System.ReadOnlySpan<byte> Hash
    {
        get => _hash
    }

    public global::System.ReadOnlySpan<byte> Content
    {
        get => _content
    }

    public override string ToString()
    {
        return @"type Query {
                            s: String
                        }";
    }
}
